### ISSUE TITLE
Usage error on discovery failure

### ISSUE THREAD

**Issue by olegrok** [tarantool/vshard#252]
vshard v0.1.15

No repro, but in production in logs we found following errors:
```
Sep 12 09:33:59 xxx-xxxxx-xx tarantool[4650]: 2020-09-12 09:33:59.235 [4650] main/160/vshard.discovery._static_router init.lua:270 W> Error during discovery fbfa11e7-1a0f-4cbe-b430-94e25df44954: Timeout exceeded
Sep 12 09:33:59 xxx-xxxxx-xx tarantool[4650]: 2020-09-12 09:33:59.235 [4650] main/160/vshard.discovery._static_router util.lua:59 E> discovery_f has been failed: builtin/box/net_box.lua:372: Usage: future:wait_result(timeout)
Sep 12 09:34:02 xxx-xxxxx-xx tarantool[4650]: 2020-09-12 09:34:02.199 [4650] main/108/lua I> Could not reach node: 10.XXX.XX.XX3:XX12 - suspect
Sep 12 09:34:02 xxx-xxxxx-xx tarantool[4650]: 2020-09-12 09:34:02.265 [4650] main/670585363/http/10.XXX.XX.XX3:XXX98 init.lua:489 W> Replica a1700f49-746d-4535-8b5b-ff273c3db8b7 is not master for replicaset 740f55e7-98b7-4d53-9421-bbc4d14f359f anymore,please update configuration!
...
Sep 12 09:34:03 xxx-xxxxx-xx tarantool[4650]: 2020-09-12 09:34:03.104 [4650] main/159/vshard.failover._static_router I> Ping error from table: 0x462ff640: perhaps a connection is down
```

I see two interesting things:
- Usage error: `discovery_f has been failed: builtin/box/net_box.lua:372: Usage: future:wait_result(timeout)`
- Bad error formatting `Ping error from table: 0x462ff640`

**Comment by Gerold103** [comment]
Do you use 1.9? Or do you use an outdated vshard version? Such an error message on the latest vshard only exists in the code, which is active for 1.9.

**Comment by olegrok** [comment]
No, it's definitely not 1.9. It's 2.3.

**Comment by Gerold103** [comment]
This was fixed in 0.1.16 - a negative timeout was passed to future:wait_result() sometimes.

**Comment by olegrok** [comment]
Seems it was fixed in https://github.com/tarantool/vshard/commit/9f67988d2515ee5c422adb1d4c8a2e454fb71721

Closing this issue as known old problem
